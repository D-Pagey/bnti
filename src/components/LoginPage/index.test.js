import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { LoginPage } from '.';

const props = {
    history: { goBack: () => {} }
};

describe('LoginPage component', () => {
    it('should render', () => {
        const { container } = render(<LoginPage {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should handle back button', () => {
        const goBack = jest.fn();
        const history = { goBack };
        const { getByTestId } = render(<LoginPage {...props} history={history} />);
        const backButton = getByTestId('backButton');

        fireEvent.click(backButton);
        expect(goBack).toHaveBeenCalled();
    });

    it.each`
        email            | password    | disabled
        ${null}          | ${null}     | ${true}
        ${'foo@bar.com'} | ${null}     | ${true}
        ${null}          | ${'foobar'} | ${true}
        ${'foo@bar.com'} | ${'foobar'} | ${false}
    `(
        'should have submit disabled ($disabled) when email is $email and password is $password',
        ({ email, password, disabled }) => {
            const { getByTestId } = render(<LoginPage {...props} />);
            const emailInput = getByTestId('emailInput');
            const passwordInput = getByTestId('passwordInput');
            const loginButton = getByTestId('loginButton');

            if (email) fireEvent.change(emailInput, { target: { value: email } });
            if (password) fireEvent.change(passwordInput, { target: { value: password } });

            if (disabled) expect(loginButton).toBeDisabled();
            else expect(loginButton).not.toBeDisabled();
        }
    );

    it.each(['email', 'password'])('should handle %s change', (type) => {
        const value = 'foobar';
        const { getByTestId } = render(<LoginPage {...props} />);
        const input = getByTestId(`${type}Input`);

        expect(input.value).toEqual('');
        fireEvent.change(input, { target: { value } });
        expect(input.value).toEqual(value);
    });
});
