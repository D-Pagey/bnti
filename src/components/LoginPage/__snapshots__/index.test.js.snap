// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component should render 1`] = `
<form>
  <input
    data-testid="emailInput"
    type="email"
    value=""
  />
  <input
    data-testid="passwordInput"
    type="password"
    value=""
  />
  <button
    data-testid="loginButton"
    disabled=""
    type="submit"
  >
    Login
  </button>
  <button
    data-testid="backButton"
    type="button"
  >
    Go Back
  </button>
</form>
`;
